recipe-client-addon/test/browser/browser_EnvExpressions.js

```diff
@@ -65,4 +66,23 @@ add_task(function* () {
   is(val, true, "Bucket sample returns true for a known sample");
   val = yield EnvExpressions.eval('["test-4"]|bucketSample(0, 5, 10)');
   is(val, false, "Bucket sample returns false for a known sample");
+
+  // Test that userId is available
+  val = yield EnvExpressions.eval("normandy.userId");
+  ok(Utils.UUID_REGEX.test(val), "userId available");
+
+  // test that it pulls from the right preference
+  Preferences.set("extensions.shield-recipe-client.user_id", "fake id");
```

Comment:
(ie then you can omit the additional `Preferences.set` call at the end, and we guarantee we restore a previous value, and we guarantee the pref will be reset even if the test times out / fails / does whatever.)